docs/observability.md

```diff
@@ -0,0 +1,69 @@
+# OpenTelemetry Service Observability
+
+## Goal
+
+The goal of this document is to have a comprehensive description of observability of the Service and changes needed to achieve observability part of our [vision](vision.md).
+
+## What Needs Observation
+
+The following elements of the Service need to be observable.
+
+### Current Values
+
+- Resource consumption: CPU, RAM (in the future also IO - if we implement persistent queues) and any other metrics that may be available to Go apps (e.g. garbage size, etc).
+- Receiving data rate, broken down by receivers and by data type (traces/metrics).
+- Exporting data rate, broken down by exporters and by data type (traces/metrics).
+- Data drop rate due to throttling, broken down by data type.
+- Data drop rate due to invalid data received, broken down by data type.
+- Current throttling state: Not Throttled/Throttled by Downstream/Internally Saturated.
+- Incoming connection count, broken down by receiver.
+- Incoming connection rate (new connections per second), broken down by receiver.
+- In-memory queue size (in bytes and in units).
+- Persistent queue size (when supported).
+- End-to-end latency (from receiver input to exporter output). Note that with multiple receivers/exporters we potentially have NxM data paths, each with different latency (plus different pipelines in the future), so realistically we should likely expose the average of all data paths (perhaps broken down by pipeline).
+- Latency broken down by pipeline elements (including exporter network roundtrip latency for request/response protocols).
+
+“Rate” values must reflect the average rate of the last 10 seconds. Rates must exposed in bytes/sec and units/sec (e.g. spans/sec).
+
+Note: some of the current values may be calculated as derivatives of cumulative values in the backend, so it is an open question if we want to expose them separately or no.
+
+### Cumulative Values
+
+- Total received data, broken down by receivers and by data type (traces/metrics)
+- Total exported data, broken down by exporters and by data type (traces/metrics)
+- Total dropped data due to throttling, broken down by data type.
+- Total dropped data due to invalid data received, broken down by data type.
+- Total incoming connection count, broken down by receiver.
+- Uptime since start.
+
+### Aggregate Values
+
+For the following metrics we want to track min, average and max since start: TBD.
```

Comment:
The point is more to have a consistent set of histograms applicable to specific kind of metrics.

Reply:
Histogram calculation can be also more expensive. Do we want it to be done in the service? With histograms we are really moving into the realm of full-blown metric backends, perhaps crossing the line of what should be done in-house by the service itself.

Reply:
Agree with @pjanotti. IMO calculating histogram aggregation is not essentially different from calculating cumulative counts/bytes/etc. It may be more difficult to figure out the rate of histograms but I think that's the case where we can defer it to the backend.

Reply:
I removed this section for now since it is incomplete. We can add it in a future PR.